tests/framework/e2e/etcd_spawn.go

```diff
@@ -15,10 +15,27 @@
 package e2e
 
 import (
+	"strings"
+
 	"go.etcd.io/etcd/pkg/v3/expect"
 	"go.uber.org/zap"
 )
 
 func SpawnCmd(args []string, envVars map[string]string) (*expect.ExpectProcess, error) {
 	return SpawnCmdWithLogger(zap.NewNop(), args, envVars)
 }
+
+// SpawnCmdWithJsonPrinter spawns an etcdctl process with provided command line arguments and environment variables
+// with json printer output and capture header line as output.
+func SpawnCmdWithJsonPrinter(args []string, envVars map[string]string) (string, error) {
+	args = append(args, "-w", "json")
+	cmd, err := SpawnCmd(args, envVars)
+	if err != nil {
+		return "", err
+	}
+	return cmd.ExpectFunc(func(txt string) bool {
+		// lease response does not have header substring
+		// member add response does not have revision substring
+		return strings.Contains(txt, "header") || strings.Contains(txt, "revision")
```

Comment:
I consider using or here a workaround for tests, however I think more important here is to find out why we have such inconsistency and consider fixing it. Could you give exact examples of `Response` protos in https://github.com/etcd-io/etcd/blob/main/api/etcdserverpb/rpc.proto that are missing some fields?

Reply:
Yeah, let me raise a issue instead of fixing it in the tests migration PR.

Reply:
It's not grpc client implementation missed some fields but client v3. 

Examples:
1. `header` substring is missing because of embedding.
https://github.com/etcd-io/etcd/blob/b7be91f98b7a7a6202ed9e9bc37b692d484585a8/client/v3/lease.go#L71-L75
2. `revision` is not populated in
https://github.com/etcd-io/etcd/blob/b7be91f98b7a7a6202ed9e9bc37b692d484585a8/server/etcdserver/api/v3rpc/member.go#L108-L110

Reply:
Thanks for raising this. 

For lease, we need some big change on this. There is an [issue](https://github.com/etcd-io/etcd/issues/13915) discussing this.  The problem mentioned here is lease response indeed has the header info, but it isn't included under the "header" section. For example,

```
$ etcdctl lease list -w json | jq
{
  "cluster_id": 14841639068965180000,
  "member_id": 10276657743932975000,
  "revision": 1,
  "raft_term": 2,
  "leases": []
}

$ etcdctl  get k1 -w json | jq
{
  "header": {
    "cluster_id": 14841639068965180000,
    "member_id": 10276657743932975000,
    "revision": 1,
    "raft_term": 2
  }
}
```

With regard to the missing "revision" in the memer related commands, I think we might need to fix it. Feel free to raise a PR to send out for review. 
```
$ etcdctl member list  -w json | jq
{
  "header": {
    "cluster_id": 14841639068965180000,
    "member_id": 10276657743932975000,
    "raft_term": 2
  },
  "members": [
    {
      "ID": 10276657743932975000,
      "name": "default",
      "peerURLs": [
        "http://localhost:2380"
      ],
      "clientURLs": [
        "http://localhost:2379"
      ]
    }
  ]
}
```

Reply:
> The problem mentioned here is lease response indeed has the header info, but it isn't included under the "header" section

Exactly. I think may be a json tag added to the LeaseLeasesResponse struct should help. It's a relatively small change.

Reply:
Technically speaking, it's a small change. But it has impact on user experience. So we should be cautious about this.  Let's track this in the issue I mentioned above.

Feel free to fix the mission revision in member related commands for now.

Reply:
Yeah, 100% agree. I just realized it like kube-apiserver will be impacted by this small change so I dropped the drafted PR.